docs: add nginx unit section (#2615)
litestar-org/litestar: docs/usage/deployment/nginx-unit.rst

Reviewer: Same as above on moving to Topics section

Author: Done

NGINX Unit
==========

To run litestar using `nginx-unit <https://unit.nginx.org/>`_

Install nginx-unit
------------------

> 1. To install nginx-unit, refers to the `official documentation <https://unit.nginx.org/installation/>`_

Reviewer: ```suggestion
To install ``nginx-unit``, refer to the `official documentation <https://unit.nginx.org/installation/>`_
```

.. tab-set::

>    .. tab-item:: macOS (Brew)

Reviewer: ```suggestion
   .. tab-item:: macOS (`Brew <https://brew.sh/>`_)
```

        .. literalinclude:: /examples/deployment/nginx-unit/install-macos.sh
            :language: sh

   .. tab-item:: Ubuntu

        To be done

Start the process, replace `user` by your system user.

.. code-block:: sh

    unitd --user <user>

3. Create a `run.py` file containing the reference of your litestar app

.. literalinclude:: /examples/todo_app/hello_world.py
    :language: python
    :caption: run.py

Configuration
-------------

Create a file called `unit.json`, put it at the root of the your project

.. literalinclude:: /examples/deployment/nginx-unit/unit.json
    :language: json
    :caption: unit.json

Listeners
+++++++++

To accept requests, add a listener object in the config/listeners API section; the object’s name can be:

> - A unique IP socket: 127.0.0.1:80, [::1]:8080
> - A wildcard that matches any host IPs on the port: *:80
> - On Linux-based systems, abstract UNIX sockets can be used as well: unix:@abstract_socket.

Reviewer: ```suggestion
- A unique IP socket: ``127.0.0.1:80``, ``[::1]:8080``
- A wildcard that matches any host IPs on the port: ``*:80``
- On Linux-based systems, abstract UNIX sockets can be used as well: ``unix:@abstract_socket``.
```
sphinx will add a link to that `*`

+------------------------+-------------------+----------------------------------------------------------------------------------------------------------------------------------------------------+
| ``processes``          | 1                 | Integer or object; integer sets a static number of app processes, and object options `max`, `spare`, and `idle_timeout` enable dynamic management. |
|                        |                   |                                                                                                                                                    |
|                        |                   | The default is 1.                                                                                                                                  |
+------------------------+-------------------+----------------------------------------------------------------------------------------------------------------------------------------------------+

Configuration update
--------------------

> To update the nginx-unit service already running, use ``PUT`` method to send the ``unit.json`` file on the `/config` endpoint

Reviewer: ```suggestion
To update the nginx-unit service already running, use ``PUT`` method to send the ``unit.json`` file on the ``/config`` endpoint
```
